nextjs-app/components/common/TagFilter/TagFilter.tsx

```diff
@@ -0,0 +1,45 @@
+"use client";
+
+import { IFilterOption } from "@/constants/filter-options";
+import { default as FilterAnchor } from "@/public/images/filter-anchor.svg";
+
+interface ITagFilterProps {
+  filterOptions: IFilterOption[];
+  selectedFilter: IFilterOption["key"];
```

Comment:
```suggestion
  selectedFilter: string;
```
Actually, it is just string, right ? Maybe we only need to use `IFilterOption["key"]` when the key is a union of static string such as `'all' | 'lecture' | 'payment'...`